packages/reader/src/allure1/index.ts

```diff
@@ -51,19 +51,20 @@ const arrayTags: Set<string> = new Set([
   "test-suite.labels.label",
   "test-suite.test-cases.test-case",
   "test-suite.test-cases.test-case.labels.label",
-  "test-suite.test-cases.test-case.steps.step",
-  "test-suite.test-cases.test-case.attachments.attachment",
   "test-suite.test-cases.test-case.parameters.parameter",
-  "test-suite.test-cases.test-case.steps.step.attachments.attachment",
 ]);
+const arrayTagPatterns = [
+  /^test-suite\.test-cases\.test-case(?:\.steps\.step)+$/,
+  /^test-suite\.test-cases\.test-case(?:\.steps\.step)*\.attachments\.attachment$/,
+];
 
 const xmlParser = new XMLParser({
   parseTagValue: false,
   ignoreAttributes: false,
   attributeNamePrefix: "",
   removeNSPrefix: true,
   allowBooleanAttributes: true,
-  isArray: (tagName, jPath) => arrayTags.has(jPath),
+  isArray: (tagName, jPath) => arrayTags.has(jPath) || arrayTagPatterns.some((p) => p.test(jPath)),
```

Comment:
Maybe check tag names instead of paths? `attachment`, `step`, `parameter`, 'label', 'test-case' tags are always arrays